src/instance_info.rs

```diff
@@ -0,0 +1,191 @@
+				msg: "Error: Invalid info extension".into(),
+				prefs: Preferences::new(&req),
+				url: req.uri().to_string(),
+			}
+			.render()
+			.unwrap();
+			Response::builder().status(404).header("content-type", "text/html; charset=utf-8").body(error.into())
+		}
+	};
+	response.map_err(|err| format!("{err}"))
+}
+
+fn info_json() -> Result<Response<Body>, Error> {
+	let body = serde_json::to_string(&*INSTANCE_INFO).unwrap_or("Error serializing JSON.".into());
+	Response::builder().status(200).header("content-type", "application/json").body(body.into())
+}
+
+fn info_yaml() -> Result<Response<Body>, Error> {
+	let body = serde_yaml::to_string(&*INSTANCE_INFO).unwrap_or("Error serializing YAML.".into());
+	// https://github.com/ietf-wg-httpapi/mediatypes/blob/main/draft-ietf-httpapi-yaml-mediatypes.md
+	Response::builder().status(200).header("content-type", "application/yaml").body(body.into())
+}
+
+fn info_txt() -> Result<Response<Body>, Error> {
+	Response::builder()
+		.status(200)
+		.header("content-type", "text/plain")
+		.body(INSTANCE_INFO.to_string(StringType::Raw).into())
+}
+fn info_html(req: Request<Body>) -> Result<Response<Body>, Error> {
+	let message = MessageTemplate {
+		title: String::from("Instance information"),
+		body: INSTANCE_INFO.to_string(StringType::Html),
+		prefs: Preferences::new(&req),
+		url: req.uri().to_string(),
+	}
+	.render()
+	.unwrap();
+	Response::builder().status(200).header("content-type", "text/html; charset=utf8").body(message.into())
+}
+#[derive(Serialize, Deserialize, Default)]
+pub(crate) struct InstanceInfo {
+	crate_version: String,
+	git_commit: String,
+	deploy_date: String,
+	compile_mode: String,
+	deploy_unix_ts: i64,
+	config: Config,
+}
+
+impl InstanceInfo {
+	pub fn new() -> Self {
+		Self {
+			crate_version: env!("CARGO_PKG_VERSION").to_string(),
+			git_commit: env!("GIT_HASH").to_string(),
+			deploy_date: OffsetDateTime::now_local().unwrap_or_else(|_| OffsetDateTime::now_utc()).to_string(),
+			#[cfg(debug_assertions)]
+			compile_mode: "Debug".into(),
+			#[cfg(not(debug_assertions))]
+			compile_mode: "Release".into(),
+			deploy_unix_ts: OffsetDateTime::now_local().unwrap_or_else(|_| OffsetDateTime::now_utc()).unix_timestamp(),
+			config: CONFIG.clone(),
+		}
+	}
+	fn to_table(&self) -> String {
+		let mut container = Container::default();
+		let convert = |o: &Option<String>| -> String { o.clone().unwrap_or("<span class=\"unset\"><i>Unset</i></span>".to_owned()) };
+		if let Some(banner) = &self.config.banner {
+			container.add_header(3, "Instance banner");
+			container.add_raw("<br />");
+			container.add_paragraph(banner);
+			container.add_raw("<br />");
+		}
+		container.add_table(
+			Table::from([
+				["Crate version", &self.crate_version],
+				["Git commit", &self.git_commit],
+				["Deploy date", &self.deploy_date],
+				["Deploy timestamp", &self.deploy_unix_ts.to_string()],
+				["Compile mode", &self.compile_mode],
+			])
+			.with_header_row(["Settings"]),
```

Comment:
At least on Firefox 108.0.2, the table header spans only the first column and not the entire row. This might be a limitation/bug of the crate and not a fault on your part.

![image](https://user-images.githubusercontent.com/13266270/212814802-efff75ba-2279-4017-85ef-f396d512281b.png)

Reply:
Yep - this is a limitation I can't directly change but can be patched around with a replace call.